graphql/src/runner.rs

```diff
@@ -106,9 +114,33 @@ where
         Ok(exts)
     }
 
+    /// Check if the subgraph state differs from `state` now in a way that
+    /// would affect a query that looked at data as fresh as `latest_block`.
+    /// If the subgraph did change, return the `Err` that should be sent back
+    /// to clients to indicate that condition
+    fn deployment_changed(
+        &self,
+        state: DeploymentState,
+        latest_block: u64,
+    ) -> Result<(), QueryExecutionError> {
+        if *GRAPHQL_ALLOW_DEPLOYMENT_CHANGE {
+            return Ok(());
+        }
+        let new_state = self.store.deployment_state_from_id(state.id.clone())?;
+        if state.reorg_count != new_state.reorg_count {
```

Comment:
Couldn't we assert here that `new_state.reorg_count >= state.reorg_count` and therefore this check can be `state.reorg_count < new_state.reorg_count`. The test would trip this assert, as said in the other comment.

Reply:
Yep, changed that.